account_invoice_refund_link/models/account_invoice.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2004-2011 Pexego Sistemas Informáticos. (http://pexego.es)
+# Copyright 2016 Antonio Espinosa <antonio.espinosa@tecnativa.com>
+# Copyright 2014-2018 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    refund_invoice_ids = fields.One2many(
+        "account.move", "reversed_entry_id", string="Refund Invoices", readonly=True
+    )
+
+    @api.model
+    def _reverse_move_vals(self, default_values, cancel=True):
+        move_vals = super()._reverse_move_vals(default_values, cancel)
+        if move_vals["type"] in ("out_refund", "in_refund"):
+            refund_lines_vals = move_vals.get("line_ids", [])
+            for i, line in enumerate(self.line_ids):
+                if i + 1 > len(refund_lines_vals):  # pragma: no cover
+                    # Avoid error if someone manipulate the original method
+                    break
+                refund_lines_vals[i][2]["origin_line_ids"] = [(6, 0, line.ids)]
+        return move_vals
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.move.line"
+
+    origin_line_ids = fields.Many2many(
```

Comment:
you should add migration scripts for these two many2many tables. I can help you if you want :)

Reply:
Yes, migration scripts must be added, but I'm not strong doing that.

Reply:
If I need your help I will tell you

Reply:
Hi @MiquelRForgeFlow , yes, I would appreciate your help, i need it. Thank you.

Reply:
And why not leaving `many2one` in refunds and `one2many` in origin invoices? I think the data structure is enough, as one refund line can only proceed from one invoice line, isn't it?

Reply:
I agree, so I did it, check it out. Thanks.